x/oracle/types/validatorPerNetwork.go

```diff
@@ -0,0 +1,68 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// UpdateValidator reset validator's power
+func (list *ValidatorWhiteList) UpdateValidator(validator sdk.ValAddress, power uint32) {
+	list.GetWhiteList()[validator.String()] = power
+}
+
+// GetValidatorPower return validator's power
+func (list *ValidatorWhiteList) GetValidatorPower(validator sdk.ValAddress) uint32 {
+	if list.ContainValidator(validator) {
+		return list.GetWhiteList()[validator.String()]
+	}
+
+	return 0
+}
+
+// ContainValidator return if validator in the map
+func (list *ValidatorWhiteList) ContainValidator(validator sdk.ValAddress) bool {
+	_, ok := list.GetWhiteList()[validator.String()]
+	return ok
+}
+
+// GetPowerRatio return the power ratio of input validator address list
+func (list *ValidatorWhiteList) GetPowerRatio(claimValidators map[string][]sdk.ValAddress) (string, float64, float64) {
+	var totalPower = uint32(0)
+	for _, value := range list.GetWhiteList() {
+		totalPower += value
+	}
+
+	var totalClaimPower = uint32(0)
+	var highestClaimPower = uint32(0)
+	var highestString = ""
+
+	for claim, validatorAddresses := range claimValidators {
+		claimPower := uint32(0)
+		for _, address := range validatorAddresses {
+			claimPower += list.GetValidatorPower(address)
+		}
+
+		totalClaimPower += claimPower
+		if claimPower > highestClaimPower {
+			highestClaimPower = claimPower
+			highestString = claim
+		}
+	}
+
+	return highestString, float64(highestClaimPower) / float64(totalPower), float64(totalClaimPower-highestClaimPower) / float64(totalPower)
```

Comment:
Why are we returning three separate things? Also why do we need to return a string?

Reply:
the string represent the prophecy with highest vote.

Reply:
other two numbers used separately to judge if the prophecy is finalized or failed.

Reply:
This will be addressed in the new PR around how we create prophecy ID's.